src/models/model.cpp

```diff
@@ -1920,6 +1942,12 @@ void model::replace_layer(OwningLayerPtr&& new_layer, std::string const& old_lay
   parent.replace_child_layer(new_layer, parent.find_child_layer_index(l));
   new_layer->add_parent_layer(m_layers[old_layer_index-1]);
```

Comment:
This may do something weird if the layer graph isn't sequential:
```suggestion
  new_layer->add_parent_layer(l.get_parent_layer_pointer(0));
```

Reply:
This is an important fix!